bin/dar

```diff
@@ -0,0 +1,279 @@
+
+  run(
+    [
+      'docker', 'run',
+      '--detach',
+      '--name', $name,
+      '--mount', "type=bind,src=$ABS_CWD,dst=/srv/cyrus-imapd",
+      ($opt->keep ? () : '--rm'),
+      $class->_requested_image($opt),
+      qw( cyd idle )
+    ],
+    emptyref(),
+    \my $container_id,
+  );
+
+  chomp $container_id;
+  say "✅ Container started, id: $container_id";
+
+  Process::Status->assert_ok("❌ Starting idle container");
+
+  my $container = $class->_existing_container;
+
+  unless ($container) {
+    # This is another one of those "should never happen" things…
+    die "❌ The container was started, but now can't be found!\n"
+  }
+
+  # We need the git-version.sh program to work, which means that "git describe"
+  # needs to work in the container's git repo, but it will be running as root,
+  # so git will complain about mismatched ownership unless we mark this
+  # directory safe. -- rjbs, 2024-12-27
+  run([
+    qw( docker exec ), $container->{ID},
+    qw( git config --global --add safe.directory /srv/cyrus-imapd ),
+  ]);
+
+  Process::Status->assert_ok("❌ Fixing git permissions in container");
+
+  return $class->_existing_container;
+}
+
+sub do_prune ($class, $opt, $args) {
+  @$args && die "❌ You can't supply a command to run with --prune.\n";
+  $opt->image && die "❌ You can't use --image with --prune.\n";
+
+  my $container = $class->_existing_container;
+
+  unless ($container) {
+    say "✅ Nothing to clean up.";
+    return;
+  }
+
+  run(
+    [ qw( docker inspect ), $container->{ID} ],
+    emptyref(),
+    \my $inspect_json,
+  );
+
+  Process::Status->assert_ok("❌ Inspecting stopped container");
+
+  my $inspect = decode_json($inspect_json);
+  my $autoremove = $inspect->[0]{HostConfig}{AutoRemove};
+
+  run(
+    [ qw( docker container stop ), $container->{ID} ],
+    emptyref(),
+    emptyref(),
+  );
+
+  Process::Status->assert_ok("❌ Stopping existing container");
+
+  say "✅ Container stopped.";
+
+  unless ($autoremove) {
+    run([ qw( docker container rm ), $container->{ID} ]);
+    Process::Status->assert_ok("❌ Removing stopped container");
+  }
+}
+
+sub do_exec ($class, $opt, $args) {
+  my $container = $class->_existing_container;
+
+  unless ($container && $container->{State} eq 'running') {
+    $container = $class->do_start($opt, $args);
+  }
+
+  unless ($container) {
+    # I don't think this should ever happen…
+    die "❌ Couldn't get a container to run that in, sorry!\n";
+  }
+
+  if ($container->{State} ne 'running') {
+    # I don't think this should ever happen either...
+    die "❌ Existing container isn't running, you should prune...\n";
+  }
+
+  if ($container->{Command} ne q{"cyd idle"}) {
+    # I don't think this should ever happen either...
+    die "❌ Weird: existing container isn't running cyd idle.  Giving up.\n";
+  }
+
+  say "⏳ Executing command in container $container->{ID}...";
+
+  my @to_run = @$args && $args->[0] eq 'run'  ? @$args
```

Comment:
This passes `run` down to the container which doesn't exist I think?

Maybe you want
```
-  my @to_run = @$args && $args->[0] eq 'run'  ? @$args
+  my @to_run = @$args && $args->[0] eq 'run'  ? @$args[1..$#$args]
```

Reply:
Yes, thanks for the catch.